client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -53,20 +53,22 @@ class PostRelativeTime extends PureComponent {
 				nextDay: this.props.translate( '[tomorrow at] LT', {
 					comment: 'LT refers to time (eg. 18:00)',
 				} ),
-				sameElse: this.props.translate( 'll [at] LT', {
-					comment:
-						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" is translated',
-				} ),
+				sameElse:
+					this.props.translate( 'll [at] LT', {
+						comment:
+							'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" is translated',
+					} ) ?? '',
 			} );
 		} else {
 			if ( Math.abs( now.diff( this.getTimestamp(), 'days' ) ) < 7 ) {
 				return timestamp.fromNow();
 			}
 
-			const sameElse = this.props.translate( 'll [at] LT', {
-				comment:
-					'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
-			} );
+			const sameElse =
+				this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+				} ) ?? '';
```

Comment:
I think it might be better to keep the default untranslated string so there will at least be some date if the translation hasn't loaded yet or fails to load for some reason. I'm not convinced, but folks from @Automattic/i18n might be able to confirm.

In any case, this should fix the warning for the time being and we can follow up with another PR to improve the situation if we reconsider.

Reply:
From what I understand, I don't think there's an underlying issue. `translate()` is expected to return literally anything - an empty string, `null`, and basically any React node. The problem was that this polymorphic behavior wasn't considered by whoever used `translate()` inside `timestamp.calendar` - they likely expected `translate()` to just receive a string and return a string - that's obviously not the case. So by defaulting to a string, you're going around it, although I'd recommend falling back to the original string (`ll at LT` - notice I'm removing the brackets), because that will at least pass some date format to `timestamp.calendar` and it will still be able to function; while with an empty string the end result will be undetermined.